### DISCUSSION TITLE
how to make it working on Radeon 7 5800H?

### DISCUSSION THREAD

**Discussion by homonto** [isc30/ryzen-gpu-passthrough-proxmox#18] [General]
So I tried this one on my Beelink SER 5 Pro but apparently all is ok except there is no GPU in windows ;-)
Which part is different from 5800 and 7000?
Of course I installed AMD drivers for 5800H but actually it says:
<img width="1310" alt="Screenshot 2024-02-05 at 20 40 10" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/46562447/5f5449df-8b6c-4fcd-9668-3813c933107c">

<img width="794" alt="Screenshot 2024-02-05 at 20 41 52" src="https://github.com/isc30/ryzen-7000-series-proxmox/assets/46562447/d3b52407-14e3-49a8-ad12-34cccc2d78d9">

**Comment by augiem** [comment]
I got it working on a 5700U using OVMF (UEFI) by extracting AMDGopDriver.efi and converting it to a rom from my motherboard bios rom. This was the key for me. Tried everything under the sun. The AMDGopDriver provided in this repo didn't work, and neither did one provided by someone else with the same CPU but a different motherboard.

Most of the steps described in the main readme are required, except use OVMF (UEFI) instead of SeaBios.

This assumes you have access to Windows to extract the AMDGopDriver rom:

- [Download EDK II Base Tools EfiRom.exe from here](https://github.com/tianocore/edk2-BaseTools-win32/blob/master/EfiRom.exe).
- [Download UEFI Bios Updater (UBU) from here](https://winraid.level1techs.com/t/tool-guide-news-uefi-bios-updater-ubu/30357).
- Download the latest BIOS ROM for your motherboard.

Extract AMDGopDriver and VBIOS:
- Run ubu.bat and select motherboard bios .bin file.
- Select 2 - Video OnBoard
- Select S - Share Files
- The driver will be extracted to: Extracted/GOP/.../AMDGopDriver.efi
- The vbios file will be extracted to: Extracted/VBIOS/.../vbios_####.dat

Convert AMDGopDriver.efi to rom:
- Using EfiRom.exe downloaded above, run the command, replacing #### with the product id of VGA compatible adapter obtained earlier when setting up Proxmox and "path/to/" with the path to the extracted efi file: (Note: It also seems to work to use the product id suffix from the vbios_####.dat file instead...)
    EfiRom.exe -f 0x1002 -i 0x#### -e path/to/AMDGopDriver.efi
- This will create AMDGopDriver.rom in the same directory.

Copy files:
    Copy AMDGopDriver.rom and vbios_####.dat to the Proxmox host in the directory /usr/share/kvm/

Edit your /etc/pve/qemu-server/###.conf and append the  romfile=vbios_####.dat file to the GPU PCIE device and romfile=AMDGopDriver.rom to the audio device PCIE entry. (See the main readme for info about editing this file.)

Without the correct AMDGopDriver, I was unable to ever get any signal at all over HDMI in either CSM or UEFI mode. With the correct GOP driver, the moment I start the VM, I get a signal showing the Proxmox logo. I've found that to be the key. If the display output just shows text from the initial Proxmox boot log and doesn't disappear or change when you launch the VM, nothing you do inside Windows will ever make it work because the GPU passthrough did not work.

**Reply by isc30** [reply]
Hi @augiem, happy to see that you got it working. Can you share your AMDGopDriver.efi together with the hardware model?

I can upload it to the repo

**Reply by qs-data** [reply]
Hi @augiem... just wanted to say thanks, I was beating my head against the wall trying to get the vbios files i needed and this was the ticket (my firmware tables didn't contain the VFCT file assumed in the vbios script in this git so it errored out).

I was about to give up but this was the ticket.

<cue drumroll....>

@isc30 ... big thanks to you for this wonderful guide!!!! I've used this and have seen may references to it in the proxmox forums.
My extracted files that appear to be working for my old Asus Prime B350 Plus and 5500GT:
[AMDGopDriver_and_vbios_5500GT.zip](https://github.com/user-attachments/files/17535445/AMDGopDriver_and_vbios_5500GT.zip)

**Comment by homonto** [comment]
ha, I got rid of 5800H and bought another machine - this time MinisForum UM790 Pro Ryzen 9 7940HS
I would love to use this iGPU but it is horrible job 
and NO, I DON'T have Windows

**Reply by augiem** [reply]
I guess my eyes must be deceiving me. Those two pictures you posted above sure look like Windows...

**Comment by homonto** [comment]
yes, windows INSIDE proxmox
what I am saying is I don't have windows on bare metal machine ;-)
for the purpose of running windows on bare metal and extracting the ROM

**Reply by augiem** [reply]
There is no need to have Windows bare metal. All Windows is used for is to extract the rom from the motherboard bios, which is a file you download from the internet. The extraction does not need to be run on the target machine.

**Reply by dmuiX** [reply]
oh man...your right... i thought i need my real hardware but yeah for this step this is not necessary...

but for the other methode to extract the vbios using cpuz!

**Comment by homonto** [comment]
ok, then I did not get it

**Comment by augiem** [comment]
> Hi @augiem, happy to see that you got it working. Can you share your AMDGopDriver.efi together with the hardware model?
> 
> I can upload it to the repo

Sure! I've attached both files.

Thanks for maintaining this. I was very glad to find it and see that AMD passthrough is possible after all after having been lost in a sea of old / incomplete information and failed attempts.

I think it would be very helpful for others to add the instructions I wrote above on how to extract your own vbios and AMDGopDriver.rom from the motherboard bios to the repo so everything is centralized. I had no idea where the AMDGopDriver.rom in your repo came from, how to get a different one, or even if it was a motherboard-specific thing or not until I dug around and ended up learning what tools I needed and the process to extract it by watching a [Chinese video posted to YouTube ](https://www.youtube.com/watch?v=BElSsyLSX5c)(No, I don't speak Chinese :). It turned out to be the key to success for me. I was unable to get it working even in CSM (SeaBios) mode without using an AMDGopDriver, but in UEFI mode it works as long as I have the correct one (I haven't tried CSM mode with the new driver.)

[VbiosAndGopDriver_GMKTecNucBoxM5_Ryzen5700U.zip](https://github.com/isc30/ryzen-7000-series-proxmox/files/14452563/VbiosAndGopDriver_GMKTecNucBoxM5_Ryzen5700U.zip)

**Reply by isc30** [reply]
Awesome! thanks a lot, i just uploaded your files 😄 

In the repo, there is [the tools folder](https://github.com/isc30/ryzen-7000-series-proxmox/tree/main/Tools) where this could be documented, I didn't spend too much time trying to document it tbh, just saved the tools there just in case I needed to repeat the process haha

I accept PRs, if you want to improve the README.md file in there 🎉

**Comment by augiem** [comment]
> Awesome! thanks a lot, i just uploaded your files 😄
> 
> In the repo, there is [the tools folder](https://github.com/isc30/ryzen-7000-series-proxmox/tree/main/Tools) where this could be documented, I didn't spend too much time trying to document it tbh, just saved the tools there just in case I needed to repeat the process haha
> 
> I accept PRs, if you want to improve the README.md file in there 🎉

Awesome! Oh wow. I didn't even know those tools were in there. D'oh! I guess I didn't look very hard. Thanks again!

**Comment by augiem** [comment]
Hmm... I think the GOP driver rom should be renamed something more generic like _5700U or something rather than the name of my motherboard. Someone else with a different system was able to get their 5700U working using the same instructions I gave them to extract the rom. I tested their ROM in my system and it worked. The file size was identical, and there were only 1 or 2 bytes different between the files, but I believe that is due to the fact that he extracted the rom passing in the product id of the audio device rather than the VGA device. It seems to pretty much ignore the product id I guess. So I think it's fairly safe to say that the GOP driver should work for any 5700U.

**Reply by isc30** [reply]
thanks, renamed :)

**Comment by dmuiX** [comment]
Thanks a ton for these valuable information. never found anywhere else how to extract vbios and that its necessary to have the AMDGopDriver...

**Reply by dmuiX** [reply]
For myself as a record and maybe it helps others as well: On my Ryzen_4350G for the audio device the vendor ID in windows was different than the one on Linux. And using what I found in linux actually made it so passthrough was not possible.
So to convert from AMDGopDriver.efi to AMDGopDriver.rom you definitely should use what you find in Windows as a device ID!!!

**Comment by HouCoder** [comment]
Thanks to @augiem, I successfully passed through the AMD iGPU to Windows 11 on my BeeLink SER5 Max server with PVE 8.4.5. Your comment and the YouTube tutorial were super helpful!

<img width="904" height="593" alt="image" src="https://github.com/user-attachments/assets/bcc0557e-d13b-4613-a5ee-2dfc93727cbe" />

-----

To anyone who is suffering the same issue, if you're also using BeeLink SER5 Max and your PCIe device map looks like this 👇 , you may have a try with my rom and vbios files, this will save you some time in generating them again on Windows 😄 .

```bash
root@pve-5800h:~# lspci -nn | grep -e 'AMD/ATI'
04:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Cezanne [Radeon Vega Series / Radeon Vega Mobile Series] [1002:1638] (rev c5)
04:00.1 Audio device [0403]: Advanced Micro Devices, Inc. [AMD/ATI] Renoir Radeon High Definition Audio Controller [1002:1637]
```

And here is my full VM config

```
bios: ovmf
boot: order=ide2;ide0;scsi0
cores: 8
cpu: host
efidisk0: local:101/vm-101-disk-0.qcow2,efitype=4m,pre-enrolled-keys=1,size=528K
hostpci0: 0000:04:00.0,pcie=1,x-vga=1,romfile=5800h-vbios_164C.dat
hostpci1: 0000:04:00.1,romfile=5800h-AMDGopDriver.rom
ide0: local:iso/virtio-win-0.1.266.iso,media=cdrom,size=707456K
ide2: local:iso/Win11_24H2_Chinese_Simplified_x64.iso,media=cdrom,size=5692426K
machine: pc-q35-9.2+pve1
memory: 8192
meta: creation-qemu=9.2.0,ctime=1753496730
name: Win11
net0: virtio=BC:24:11:FA:6E:B0,bridge=vmbr0,firewall=1
numa: 0
ostype: win11
scsi0: local:101/vm-101-disk-1.qcow2,iothread=1,size=128G
scsihw: virtio-scsi-single
smbios1: uuid=14d28272-18c8-4249-801c-bc3db4157ba7
sockets: 1
tpmstate0: local:101/vm-101-disk-2.raw,size=4M,version=v2.0
usb0: host=1d6b:0104
vga: none
vmgenid: b3614640-107e-4feb-8e74-4f6069a475d1
```

[ser5-max-rom-and-vbios.zip](https://github.com/user-attachments/files/21441599/ser5-max-rom-and-vbios.zip)
